src/components/PhonesCatalog/PhoneDetailsSingle/PhoneDetailsSingle.js

```diff
@@ -46,24 +48,36 @@ const PhoneDetailsSingle = (props) => {
    return (
       <div className="phone-container">
          <div className="container">
-            <div className="phone-details-intro">
-               <Link className="phone-back" to="/phones">
-                  <p> <i className="fas fa-chevron-left" /> Back to All Phones</p>
-               </Link>
-               <h3>Page details for: <span>{isLoading ? 'Loading ...' : phoneDeviceName}</span> </h3>
-            </div>
-            <div className="phone-inner">
-               <div className="phone-details">
-                  {isLoading ? <h3>Loading...</h3> : itemProperties}
-               </div>
-               <div className="phone-img">
-                  <div className="device-purchase-details" >
-                     <p className="price-amount"> Device price: <span>{isLoading ? 'Loading' : productPrice}</span> </p>
-                     <button className="btn-addcart" onClick={addInCartHandle}>  Add to Cart </button>
+            {isLoading ?
+               <DotsLoader />
+               :
+               <div className="phone-container-content">
+                  <div className="phone-details-intro">
+                     <Link className="phone-back" to="/phones">
+                        <p> <i className="fas fa-chevron-left" /> Back to All Phones</p>
+                     </Link>
+                     <div className="phone-details-current-phone" >Page details for: <span>{isLoading ? 'Loading ...' : phoneDeviceName}</span> </div>
                   </div>
-                  {isLoading ? <h3>Loading...</h3> : <img src={randomImageSrc} alt="img" />}
-               </div>
-            </div>
+                  {errorHandle ?
+                     <NoMatch />
+                     :
+                     <div className="phone-inner">
+                        <div className="phone-details">
+                           {itemProperties}
+                        </div>
+                        <div className="phone-img">
+                           <div className="device-purchase-details" >
+                              <p className="price-amount"> Device price: <span>{isLoading ? 'Loading' : productPrice}</span> </p>
+                              <button
+                                 className="btn-addcart"
+                                 onClick={addInCartHandle}>
+                                 Add to Cart
+                           </button>
+                           </div>
+                           {isLoading ? <h3>Loading...</h3> : <img src={randomImageSrc} alt="img" />}
```

Comment:
`isLoading` is `false` only `<img src={randomImageSrc} alt="img" />` will be displayed